pkg/services/app/add.go

```diff
@@ -1023,19 +1031,21 @@ func sanitizeWegoDirectory(manifest []byte) []byte {
 	return bytes.ReplaceAll(manifest, []byte("path: ./wego"), []byte("path: .wego"))
 }
 
-func resourceKindToGVR(resourceKind ResourceKind) schema.GroupVersionResource {
-	switch resourceKind {
+func (rk ResourceKind) ToGVR() (schema.GroupVersionResource, error) {
+	switch rk {
 	case ResourceKindApplication:
-		return kube.GVRApp
+		return kube.GVRApp, nil
 	case ResourceKindSecret:
-		return kube.GVRSecret
+		return kube.GVRSecret, nil
 	case ResourceKindGitRepository:
-		return kube.GVRGitRepository
+		return kube.GVRGitRepository, nil
 	case ResourceKindHelmRepository:
-		return kube.GVRHelmRepository
+		return kube.GVRHelmRepository, nil
 	case ResourceKindHelmRelease:
-		return kube.GVRHelmRelease
+		return kube.GVRHelmRelease, nil
+	case ResourceKindKustomization:
+		return kube.GVRKustomization, nil
 	default:
-		return kube.GVRKustomization
+		return schema.GroupVersionResource{}, fmt.Errorf("no matching schema.GroupVersionResource to the ResourceKind: %s", string(rk))
```

Comment:
FYI: I think it's good that you are passing the resourceKind that caused the error.  Another approach could have been a custom error:

```
type ConversionError struct {
    kind string
}

func (e *ConversionError) Error() string {
    return fmt.Sprintf("no matching schema.GroupVersionResource to the ResourceKind: %s", string(e.kind))
}
```

The effect of this is a more clear way to check for the error consistently with `err.As()` and not having to worry about the ResourceKind changing on you.  However, this is excessive considering you aren't explicitly checking this error and handling it in a special way.

Thanks for addressing my main concern of odd side effects.